#1078 - Create custom report

HalcyonJAC opened the pull request:
## What's included?
A new way to inspect data within the dashboard and generate / save your own reports.

## Who should test?
✅ Product owner
✅ Developers
✅ UTG

## How to test?
Go to an Exercise
Click the reports tab
Click the Custom report link on the left nav

Functionality:
1. Select a column from the dropdown to show the values in a table
2. Add multiple columns, delete them by clicking and re-order by dragging.
3. Change the report type to count to view aggregated results
4. On the count report, click one of the values to add a 'where clause' which limits the results
5. Save a report by clicking Save Report at the top right of the page, give it a name. Then switch to a different exercise and try to load the report.

## Risk - how likely is this to impact other areas?
🟢 No risk - this is a self-contained piece of work

---
PREVIEW:DEVELOP
_can be OFF, DEVELOP or STAGING_

tomlovesgithub commented on src/views/Exercise/Reports/Custom.vue:
```diff
@@ -0,0 +1,555 @@
+    draggable,
+    LoadingMessage,
+  },
+  data() {
```
`isLoading` and `loadFailed` need to be added to the data here.

HalcyonJAC replied:
Hi,

This data has been added to the component.

tomlovesgithub reviewed (changes requested):
still getting console errors when testing locally ;?

warrensearle reviewed (approved):
Awesome 👍

joy-ade reviewed (commented):
feedbacks from UTG  "select a column to display" - can that be re-ordered to have application info 1st maybe as its likely the most people would filter on his on the application status

joy-ade commented:
@HalcyonJAC @JeremykJAC Can you include and export button on this report pls.

joy-ade reviewed (commented):
Feedback form UTG Can we download this report pls.

HalcyonJAC commented:
@joy-ade Hi - I've made some of the changes but I think there might be some more work to do on this later depending.

- Order of options is changed
- Added jurisdiction and location
- fixed a bug with saving reports
- added some validation on saving reports
- adding a message to show the report was saved

joy-ade reviewed (approved):
Approved by Product team and UTG
